### ISSUE TITLE
Fullscreen / borderless window without Windows 10 taskbar possible?

### ISSUE THREAD

**Issue by Stefan1200-de** [Vocaluxe/Vocaluxe#279]
I have the problem, that Vocaluxe 0.4.0.beta1-31-g2e6b388 don't open a real fullscreen window or a full sized borderless window. The Windows taskbar is always visible after starting Vocaluxe, see my attached screenshot. Other games don't have this kind of problems. What can I do here to fix this?

Using Windows 10 x64 and I have two Full HD screens.

My Graphics part of the Config.xml:
```xml
  <Graphics>
    <!--Renderer: TR_CONFIG_DIRECT3D, TR_CONFIG_OPENGL, TR_CONFIG_SOFTWARE-->
    <Renderer>TR_CONFIG_DIRECT3D</Renderer>
    <!--TextureQuality: TR_CONFIG_TEXTURE_LOWEST, TR_CONFIG_TEXTURE_LOW, TR_CONFIG_TEXTURE_MEDIUM, TR_CONFIG_TEXTURE_HIGH, TR_CONFIG_TEXTURE_HIGHEST-->
    <TextureQuality>TR_CONFIG_TEXTURE_MEDIUM</TextureQuality>
    <!--CoverSize (pixels): 32, 64, 128, 256, 512, 1024 (default: 128)-->
    <CoverSize>256</CoverSize>
    <!--Screen width and height (pixels)-->
    <ScreenW>1920</ScreenW>
    <ScreenH>1080</ScreenH>
    <ScreenAlignment>Middle</ScreenAlignment>
    <!--Screen borders (pixels)-->
    <BorderLeft>0</BorderLeft>
    <BorderRight>41</BorderRight>
    <BorderTop>0</BorderTop>
    <BorderBottom>0</BorderBottom>
    <!--AAMode: X0, X2, X4, X8, X16, X32-->
    <AAMode>X0</AAMode>
    <!--MaxFPS should be between 1..200-->
    <MaxFPS>60</MaxFPS>
    <!--VSync: TR_CONFIG_OFF, TR_CONFIG_ON-->
    <VSync>TR_CONFIG_ON</VSync>
    <!--FullScreen: TR_CONFIG_OFF, TR_CONFIG_ON-->
    <FullScreen>TR_CONFIG_ON</FullScreen>
    <!--Stretch view to full window size: TR_CONFIG_OFF, TR_CONFIG_ON-->
    <Stretch>TR_CONFIG_OFF</Stretch>
    <!--FadeTime should be between 0..3 Seconds-->
    <FadeTime>0.4</FadeTime>
  </Graphics>
```

![vocaluxe_fullscreen](https://user-images.githubusercontent.com/15168351/27545939-816b89aa-5a91-11e7-9af3-0f8c25d398ef.png)

**Comment by hendrikbl** [comment]
Try changing `<BorderRight>41</BorderRight>` to `<BorderRight>0</BorderRight>`. I'm using almost the same config and it all works fine for me. Here's my Config.xml:
```xml
<Graphics>
    <!--Renderer: TR_CONFIG_DIRECT3D, TR_CONFIG_OPENGL, TR_CONFIG_SOFTWARE-->
    <Renderer>TR_CONFIG_DIRECT3D</Renderer>
    <!--TextureQuality: TR_CONFIG_TEXTURE_LOWEST, TR_CONFIG_TEXTURE_LOW, TR_CONFIG_TEXTURE_MEDIUM, TR_CONFIG_TEXTURE_HIGH, TR_CONFIG_TEXTURE_HIGHEST-->
    <TextureQuality>TR_CONFIG_TEXTURE_HIGHEST</TextureQuality>
    <!--CoverSize (pixels): 32, 64, 128, 256, 512, 1024 (default: 128)-->
    <CoverSize>1024</CoverSize>
    <!--Screen width and height (pixels)-->
    <ScreenW>1920</ScreenW>
    <ScreenH>1080</ScreenH>
    <ScreenAlignment>Middle</ScreenAlignment>
    <!--Screen borders (pixels)-->
    <BorderLeft>0</BorderLeft>
    <BorderRight>41</BorderRight>
    <BorderTop>0</BorderTop>
    <BorderBottom>0</BorderBottom>
    <!--AAMode: X0, X2, X4, X8, X16, X32-->
    <AAMode>X32</AAMode>
    <!--MaxFPS should be between 1..200-->
    <MaxFPS>60</MaxFPS>
    <!--VSync: TR_CONFIG_OFF, TR_CONFIG_ON-->
    <VSync>TR_CONFIG_ON</VSync>
    <!--FullScreen: TR_CONFIG_OFF, TR_CONFIG_ON-->
    <FullScreen>TR_CONFIG_ON</FullScreen>
    <!--Stretch view to full window size: TR_CONFIG_OFF, TR_CONFIG_ON-->
    <Stretch>TR_CONFIG_OFF</Stretch>
    <!--FadeTime should be between 0..3 Seconds-->
    <FadeTime>0.4</FadeTime>
  </Graphics>
```

**Comment by Stefan1200-de** [comment]
@hendrikbl Unfortunately I still see the Windows taskbar, but the right sight of the screen is cut of, if I change BorderRight to 0.

By the way, in your config file the BorderRight value is also 41. :)

**Comment by hendrikbl** [comment]
> By the way, in your config file the BorderRight value is also 41. :)

Ups, tested it to see what it does and didn't change it back :D

Maybe it's got something to do with the task bar on the left. Mine is at the bottom and it works with all borders set to 0... I guess the 41 is the width of your taskbar.

**Comment by Stefan1200-de** [comment]
@hendrikbl The width of my taskbar is 62. As far as I understand, you set this values, to make sure, that the Vocaluxe screen won't be cut off. They should be 0, if Vocaluxe use a real fullscreen mode, which it not does here.

**Comment by flokuep** [comment]
Mhm. That's a bit strange... Does this changes if you use OpenGL as renderer and/or move the taskbar to the bottom (I've never tested it with taskbar on the left...)

**Comment by Stefan1200-de** [comment]
@flokuep Yes, you are right. If the taskbar is on the bottom, everything is working.  But OpenGL is not helping with this issue.

Can you fix this issue with the taskbar on the left side?

**Comment by Stefan1200-de** [comment]
This issue is still not labeled as bug. What can be done here to fix my issue?

**Comment by lukeIam** [comment]
> This issue is still not labeled as bug. What can be done here to fix my issue?

I'm not sure if it is an Vocaluxe bug - the web is full of reports of other affected applications.
e.g. https://answers.microsoft.com/en-us/windows/forum/windows_10-start/windows-10-brings-taskbar-on-top-of-fullscreen/efa61fb2-81e5-4a08-b40c-5e91d297f465

Short: I can reproduce the behavior, but I think it's a Windows bug.
Maybe there is a workaround - if you find one we're happy to implement it in Vocaluxe.

**Comment by hendrikbl** [comment]
> the web is full of reports of other affected applications.

I agree. It seems like a common problem...

@Stefan1200-de maybe you could set your taskbar to automatically hide. It's no real solution but it's better than nothing I guess

_Sent from my OnePlus ONEPLUS A3003 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Stefan1200-de** [comment]
Vocaluxe is the only application I use with this problem. No other game, and I played much in the last years, have this issue, including UltraStar DX 1.1 & 1.3 and Performous 1.1.

And I don't have the problem that the taskbar is on top of Vocaluxe, Vocaluxe is just not using that space of the taskbar, nothing is hidden by the taskbar, Vocaluxe just cut off the space from the other side! My taskbar is on the left side, Vocaluxe is not using the pixels where the taskbar is. But on the right side the Vocaluxe screen will be cut off by the pixels of the taskbar.

**Comment by lukeIam** [comment]
> And I don't have the problem that the taskbar is on top of Vocaluxe, Vocaluxe is just not using that space of the taskbar, nothing is hidden by the taskbar, Vocaluxe just cut off the space from the other side! My taskbar is on the left side, Vocaluxe is not using the pixels where the taskbar is. But on the right side the Vocaluxe screen will be cut off by the pixels of the taskbar.

Vocaluxe does get get the screen resolution (full size including taskbar) and positions the the window to the lest top corner (which is at the end of the taskbar) -> we are missing the the size of task bar on the right side.

I did a deep dive into the code and figured out the reason: Vocaluxe is not using the DirectX full screen mode for full screen. I will see what I can do here... (In a first test the taskbar problem was solved but window mode was broken instead)

**Comment by lukeIam** [comment]
@Stefan1200-de Please test the the modified version.
I build it with trail&error... but for me it's working.

**Comment by Stefan1200-de** [comment]
Thanks for testing around with this issue.

Well, using the test build the taskbar issue is gone, but now I have 30 pixel wide bar in white color on my second screen to the left.

But in general I also like the borderless window mode, which many games have (they also don't have the taskbar issue).

**Comment by lukeIam** [comment]
@Stefan1200-de 

> Well, using the test build the taskbar issue is gone, but now I have 30 pixel wide bar in white color on my second screen to the left.

I couldn't confirm this - for me it's working as expected (no bars on the other screen)
Do you have some border settings left in your `Config.xml` from testing?
```xml
    <!--Screen borders (pixels)-->
    <BorderLeft>0</BorderLeft>
    <BorderRight>0</BorderRight>
    <BorderTop>0</BorderTop>
    <BorderBottom>0</BorderBottom>
```

**Comment by Stefan1200-de** [comment]
Yes, every border is set to 0. After minimizing Vocaluxe and maximizing it again the white bar is gone. But everytime after starting Vocaluxe, I have it. I saw that Vocaluxe create a very big white box behind the DirectX output, most of the white box is on the right screen, but around 30 px are overlapped to the left screen.

**Comment by lukeIam** [comment]
> After minimizing Vocaluxe and maximizing it again the white bar is gone. But everytime after starting Vocaluxe, I have it. I saw that Vocaluxe create a very big white box behind the DirectX output, most of the white box is on the right screen, but around 30 px are overlapped to the left screen.

@flokuep Could the white box be an empty winforms window?

**Comment by marwin89** [comment]
Hi @Stefan1200-de , is this issue solved? can we close it?
